src/Microsoft.AspNetCore.Mvc.Formatters.Xml/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -59,6 +62,27 @@ public XmlDataContractSerializerInputFormatter(bool suppressInputFormatterBuffer
             WrapperProviderFactories.Add(new SerializableErrorWrapperProviderFactory());
         }
 
+        /// <summary>
+        /// Initializes a new instance of DataContractSerializerInputFormatter
+        /// </summary>
+        /// <param name="options">The <see cref="MvcOptions"/>.</param>
+        public XmlDataContractSerializerInputFormatter(MvcOptions options)
+        {
+            _options = options;
+
+            SupportedEncodings.Add(UTF8EncodingWithoutBOM);
```

Comment:
Suggestion: Up to you but you can avoid the duplication. E.g. put the common code into the parameterless constructor and have the other two constructors call that.